HHS/OPRE-OPS
# Ops 1404 fee bug squashing

## What changed

BLI Procurement Shop fees are now updated sanely to ensure that they correctly reflect the Procurement Shop choice and BLI status, whether the Agreement changes, or if the BLI changes. Testing data has been corrected so that BLIs match the Agreements they are assigned to. These were incorrect. Cleaned up tests to make everything work correctly. There was existing business logic tests for certain components already.

The majority of this problem was with the data we were loading/using. Everything matches now, so the math works out as expected.

Changing the procurement shop on the Agreement now also updates the fees on all BLIs as appropriate based on business logic.

I dried up some code for the agreements and budget line items in the backend as well.

## Issues

#1404 
#1400 
#1403 
#1381 

## How to test

pytest runs clean, as do the e2e tests. However to test this, log into the UI, and check that the math for Agreement totals is right visually, as is the math for individual BLIs.

## Screenshots

N/A

## Definition of Done Checklist


## Links

N/A
